wiki_doc

```diff
@@ -0,0 +1,190 @@
+
+#### Example Usage
+
+Here is an example of command implementation to outline arguments using `DecoderOption` and response type:
+
+```typescript
+public getdel(
+        key: GlideString,
+        options?: DecoderOption,
+    ): Promise<GlideString | null>
+```
+
+Here's a simple example demonstrating how to use `Decoder` in command API and `Buffer.from()` to encode binary data:
+
+```typescript
+expect(await client.set(key, value)).toEqual("OK");
+expect(await client.getdel(key)).toEqual(value);
+
+const valueEncoded = Buffer.from(value);
+expect(await client.set(key, value)).toEqual("OK");
+expect(await client.getdel(key, { decoder: Decoder.Bytes })).toEqual(
+    valueEncoded,
+);
+```
+
+### Transaction
+
+A transaction in Valkey Glide allows you to execute a group of commands in a single, atomic step. This ensures that all commands in the transaction are executed sequentially and without interruption. See [Valkey Transactions](https://valkey.io/topics/transactions).
+
+This is equivalent to the Valkey commands [MULTI](https://valkey.io/commands/multi/) / [EXEC](https://valkey.io/commands/exec/).
+
+#### Modes of Operation
+
+There are two primary modes for handling transactions in Glide:
+
+1. **Standalone Mode:** Use the `Transaction` class.
+2. **Cluster Mode:** Use the `ClusterTransaction` class.
+
+#### Reusing Transaction Objects
+
+Transaction objects can be reused. If you need to execute a particular group of commands multiple times, you can simply resend the same transaction object.
+
+#### Example Usage
+
+Here's a simple example demonstrating how to create and execute a transaction in standalone mode:
+
+```typescript
+// Initialize a transaction object
+const transaction = new Transaction();
+
+// Add commands to the transaction
+transaction.set("key", "value");
+transaction.select(1); // Standalone command
+transaction.get("key");
+
+// Execute the transaction
+const results = await client.exec(transaction);
+expect(results).toEqual(["OK", "OK", "value"]);
+```
+
+#### Decoder
+
+Transactions use the same decoder for all commands in the transaction. For example, if a user runs `client.exec(transaction, Decoder.Bytes)`, then all the commands in the transaction will be encoded into Bytes. The default decoder is String. There is an [issue](https://github.com/valkey-io/valkey-glide/issues/2147) to track the support of adding option to set a decoder per each command in transaction.
+
+Here's an example demonstrating how to use `Decoder` and `route` with transaction:
+
+```typescript
+let transaction = new ClusterTransaction().functionDump();
+const result = await client.exec(transaction, {
+    decoder: Decoder.Bytes,
+    route: route,
```

Comment:
I think you need a block to describe routes and possible options

Reply:
I will add the link to route and decoder before the example.